cpp/tests/iterator/offsetalator_test.cu

```diff
@@ -20,10 +20,10 @@
 
 #include <cudf/detail/offsets_iterator_factory.cuh>
 
+#include <cuda/std/optional>
 #include <thrust/binary_search.h>
 #include <thrust/gather.h>
 #include <thrust/host_vector.h>
-#include <thrust/optional.h>
```

Comment:
Looks like this include is not needed and so can be removed instead of replaced.